src/state/mod.rs

```diff
@@ -61,20 +60,22 @@ impl ClientState {
 /// Abstraction around the data store to avoid unnecessary request on client initialization.
 #[async_trait::async_trait]
 pub trait StateStore: Send + Sync {
-    /// Set up connections or check files exist to load/save state.
-    fn open(&self, path: &Path) -> Result<()>;
+    /// Signals to the `AsyncClient` if this is the first time a StateStore` has been used.
+    ///
+    /// Returns true if `StateStore` has been set up and ready to be loaded from.
+    async fn initial_use(&self) -> Result<bool>;
```

Comment:
Do we need this method? Can't we just return `Result<Option<ClientState>` to signal that no state was found in the store?

Reply:
Ooo yea that's much better